### ISSUE TITLE
Ngx-TrueHDR does not set HDR metadata -target NGX file does not display as HDR in MediaInfo 

### ISSUE THREAD

**Issue by noblemd00** [staxrip/staxrip#1509]
*Ngx-TrueHDR does not set HDR metadata -target NGX file does not display as proper HDR in MediaInfo 

The following file was encoded with Ngx-TrueHDR
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main@L4@Main
Codec ID                                 : V_MPEGH/ISO/HEVC
Duration                                 : 42 min 29 s
Bit rate                                 : 1 247 kb/s
Width                                    : 1 920 pixels
Height                                   : 872 pixels
Display aspect ratio                     : 2.2:1
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Bits/(Pixel*Frame)                       : 0.031
Stream size                              : 379 MiB (94%)
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.2020
Transfer characteristics                 : PQ
Matrix coefficients                      : BT.2020 non-constant

This video after running HDR Ingest
Video
ID                                       : 1
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main 10@L4@Main
HDR format                               : SMPTE ST 2086, HDR10 compatible

**Describe the solution you'd like**
Ngx-TrueHDR (StaxRip encode) should set metadata without having to run HDR Ingest

Whatever HDR Ingest does should run as a subroutine of the StaxRip Ngx-TrueHDR encode process

StaxRip Ngx-TrueHDR encode process should properly set HDR metadata.

 
**Describe alternatives you've considered**
Running HDR Ingest on every file re-encoded by StaxRip Ngx-TrueHDR!!!!!

**Comment by Dendraspis** [comment]
And what was your (complete) command line for `NVEncC`?

**Comment by noblemd00** [comment]
--avhw --vbr 1000 --codec h265 --preset default --output-depth 10 --aq --aq-temporal --bref-mode each --bframes 5 --lookahead 4 --multiref-l0 5 --multiref-l1 2 --strict-gop --nonrefp --vpp-ngx-truehdr  --colormatrix bt2020nc --colorprim bt2020 --transfer smpte2084 --colorrange limited --pic-struct --aud --repeat-headers --output-buf 64

**Comment by Dendraspis** [comment]
You just have to add all necessary parameters:

`--output-depth 10 --vpp-ngx-truehdr --master-display "G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(10000000,1)" --colormatrix bt2020nc --colorprim bt2020 --transfer smpte2084 --colorrange limited --max-cll "1000,300" --chromaloc 2 --aud --repeat-headers`

Th luminance values are of course just some default values, but should work for almost all cases.

**Comment by noblemd00** [comment]
I tried adding master display/ max cli items before and got an encoding error. With the previous White blooming problem and master display not working earlier, I'm wondering if a video driver (NVEnc) update fixed the earlier issues.

This time adding Master display entry worked (MediaInfo) - HDR format: SMPTE ST 2086, HDR10 compatible

**Thanks** 

Added the master display and max cli to my Ngx Event Command command line instruction.

<img width="472" alt="241017 093046" src="https://github.com/user-attachments/assets/83f509ff-1e69-4647-a62d-9d135039c43d">

If the Template name contains "Ngx" then set the following parameters...
Complete Command Line entry (other parameters set in the "Ngx" Templates or default)
--colormatrix "bt2020nc"  --colorprim "bt2020" --transfer "smpte2084" --atc-sei "unknown" --colorrange "limited" --master-display "G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(10000000,1)" --max-cll "1000,300"
